velox/common/memory/MemoryPool.h

```diff
@@ -548,9 +548,8 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
   mutable folly::SharedMutex poolMutex_;
   std::unordered_map<std::string, std::weak_ptr<MemoryPool>> children_;
 
-  friend class TestMemoryReclaimer;
   friend class MemoryReclaimer;
-  friend class core::QueryCtx;
+  friend class velox::exec::NodeMemoryReclaimer;
```

Comment:
Are there any ways we can expose the internals through API instead of making friend class?

Reply:
It is a bit complicated to expose these through internal API for this.

Reply:
I think we need to access children and lock, right? Can we expose the 2 or we have additional needs?